pkg/kubelet/kubelet.go

```diff
@@ -2122,9 +2142,19 @@ func (kl *Kubelet) cleanupOrphanedVolumes(pods []*api.Pod, runningPods []*kubeco
 				}
 
 				pdName := path.Base(refs[0])
-				err = detacher.Detach(pdName, kl.hostname)
-				if err != nil {
-					glog.Errorf("Could not detach volume %q at %q: %v", name, volumePath, err)
+				if kl.enableControllerAttachDetach {
+					// Attach/Detach controller is enabled and this volume type
+					// implments a detacher
+					uniqueDeviceName := attachdetachutil.GetUniqueDeviceName(
+						cleaner.PluginName, pdName)
+					kl.volumeManager.RemoveVolumeInUse(
+						api.UniqueDeviceName(uniqueDeviceName))
+				} else {
+					// Attach/Detach controller is disabled
+					err = detacher.Detach(pdName, kl.hostname)
+					if err != nil {
+						glog.Errorf("Could not detach volume %q at %q: %v", name, volumePath, err)
+					}
```

Comment:
I think a positive log message would be good here.  You'll want it when you're debugging :)